src/community/smart-data-loader/src/main/java/org/geoserver/smartdataloader/data/SmartDataLoaderDataAccessFactory.java

```diff
@@ -0,0 +1,356 @@
+    }
+
+    @Override
+    public final Param[] getParametersInfo() {
+        LinkedHashMap<String, DataAccessFactory.Param> map = new LinkedHashMap<>();
+        setupParameters(map);
+        return map.values().toArray(new Param[map.size()]);
+    }
+
+    @Override
+    public boolean isAvailable() {
+        try {
+            Class.forName(getDriverClassName());
+            return true;
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean canProcess(Map<String, ?> params) {
+        try {
+            Object dbType = DBTYPE.lookUp(params);
+            return DBTYPE_STRING.equals(dbType)
+                    && DataUtilities.canProcess(params, getParametersInfo());
+        } catch (Exception e) {
+            // do nothing. based on AppSchemaDataAccessFactory code
+        }
+        return false;
+    }
+
+    public String getDriverClassName() {
+        return "org.postgresql.Driver";
+    }
+
+    private void setupParameters(Map<String, Param> parameters) {
+        parameters.put(DBTYPE.key, DBTYPE);
+        parameters.put(NAMESPACE.key, NAMESPACE);
+        parameters.put(DATASTORE_NAME.key, DATASTORE_NAME);
+        parameters.put(DATASTORE_METADATA.key, DATASTORE_METADATA);
+        parameters.put(ROOT_ENTITY.key, ROOT_ENTITY);
+        parameters.put(DOMAIN_MODEL_EXCLUSIONS.key, DOMAIN_MODEL_EXCLUSIONS);
+    }
+
+    private String getFilenamePrefix(Map<String, Serializable> params) throws IOException {
+        String rootEntity = lookup(ROOT_ENTITY, params, String.class);
+        String filenamePrefix = rootEntity;
+        return filenamePrefix;
+    }
+    /**
+     * Helper method that based on parameters, builds domainmodel, generates associated mapping
+     * files and saves them in the workspace, returning the resulting DataStore.
+     */
+    private DataAccess<FeatureType, Feature> createDataStore(
+            Map<String, Serializable> params,
+            boolean hidden,
+            DataAccessMap sourceDataStoreMap,
+            final Set<AppSchemaDataAccess> registeredAppSchemaStores)
+            throws IOException {
+        // get parameters
+        URI namespace = lookup(NAMESPACE, params, URI.class);
+        String datastoreName = lookup(DATASTORE_NAME, params, String.class);
+        String excludedObjects = lookup(DOMAIN_MODEL_EXCLUSIONS, params, String.class);
+
+        // convert excluded objects in a list
+        String[] elements = {};
+        if (excludedObjects != null) {
+            elements = excludedObjects.split(",");
+        }
+        List<String> excludedObjectsList = Arrays.asList(elements);
+
+        // build domainmodel, and exclude elements based on excludedObjects list
+        DomainModel dm = buildDomainModel(params, excludedObjectsList);
+        // define filenames naming convention for documents to be saved
+        String gmlFilename = getFilenamePrefix(params) + "-gml.xsd";
```

Comment:
Please use constants for the suffixes strings.

Reply:
fixed